File: source/winmobile/NetLayer/ObjectSerializer.cs
using System;
using System.Collections.Generic;
using System.Text;
using System.Xml.Serialization;
using System.IO;
using System.Xml;

namespace NetLayer
{
    /// <summary>
    /// Clase para serializar objetos y hacerlos eviables por la red
    /// </summary>
    public class ObjectSerializer
    {
        private XmlSerializer serializer;

        /// <summary>
        /// Constructor parametrizado
        /// </summary>
        /// <param name="type">El tipo de objeto a serializar</param>
        /// <param name="extraTypes">tipos adicionales a serializar</param>
        public ObjectSerializer(Type type, Type[] extraTypes) 
        {
            if (extraTypes != null)
                serializer = new XmlSerializer(type, extraTypes);
            else
                serializer = new XmlSerializer(type);
        }

        /// <summary>
        /// Convierte un objeto en un string
        /// </summary>
        /// <param name="obj">el objeto a convertir</param>
        /// <returns>el string que representa al objeto</returns>
        public string serialize(Object obj) 
        {
            StringWriter sw = new StringWriter();
            XmlTextWriter writer = new System.Xml.XmlTextWriter(sw);
            serializer.Serialize(writer, obj);
            writer.Flush();
            writer.Close();
            return sw.ToString();
        }

        /// <summary>
        /// Convierte un string creado con serialize en un objeto
        /// </summary>
        /// <param name="obj">el objeto serializado</param>
        /// <returns>El objeto construido a partir de la serialización</returns>
        public Object unserialize(string obj) 
        {
            StringReader strxml = new StringReader(obj);
            XmlTextReader reader = new XmlTextReader(strxml);
            if(serializer.CanDeserialize(reader))
            {
                return serializer.Deserialize(reader);
            }
            else 
            {
                return null;
            }
        }
    }
}

File: source/java/hlmp/NetLayer/ObjectSerializer.java
package hlmp.NetLayer;

import java.io.StringReader;
import java.io.StringWriter;

import org.simpleframework.xml.Serializer;
import org.simpleframework.xml.core.Persister;


/**
 * Modificado para Android 2.3 por felipe.valverde.campos@gmail.com
 *
 */

public class ObjectSerializer {
	
	public String serialize(Object obj){
		StringWriter sw = new StringWriter();
		String serialized_object = "";
		Serializer serial = new Persister();
		try {
			serial.write(obj, sw);
			serialized_object = sw.toString();
		} catch (Exception e) {
			e.printStackTrace();
		}
		return serialized_object ;
	}
	
	public Object unserialize(String serialized_object, Class<?> cls){
		Serializer serializer = new Persister();
		StringReader sr = new StringReader(serialized_object);
		Object object = null;
		try {
			object = serializer.read(cls, sr);
		} catch (Exception e) {
			e.printStackTrace();
		}
		return object;
	}
}